includes/class-wc-payment-gateway-wcpay.php

```diff
@@ -617,11 +618,33 @@ public function register_scripts() {
 
 		wp_set_script_translations( 'WCPAY_CHECKOUT', 'woocommerce-payments' );
 
-		if ( ! WC()->cart->needs_payment() && is_checkout() && ! has_block( 'woocommerce/checkout' ) ) {
+	}
+
+	/**
+	 * Registers scripts necessary for the gateway, even when cart order total is 0.
+	 * This is done so that if the cart is modified via AJAX on checkout,
+	 * the scripts are still loaded.
+	 */
+	public function register_scripts_for_zero_order_total() {
+		if (
+			isset( WC()->cart ) &&
+			! WC()->cart->is_empty() &&
+			! WC()->cart->needs_payment() &&
+			is_checkout() &&
+			! has_block( 'woocommerce/checkout' )
+		) {
 			WC_Payments::get_gateway()->tokenization_script();
-			WC_Payments::get_wc_payments_checkout()->enqueue_payment_scripts();
+			if ( WC_Payments_Features::is_upe_legacy_enabled() ) {
+				wp_localize_script( 'wcpay-upe-checkout', 'wcpayConfig', WC_Payments::get_wc_payments_checkout()->get_payment_fields_js_config() );
+				wp_enqueue_script( 'wcpay-upe-checkout' );
+			} elseif ( WC_Payments_Features::is_upe_split_enabled() ) {
+				wp_localize_script( 'wcpay-upe-checkout', 'wcpay_upe_config', WC_Payments::get_wc_payments_checkout()->get_payment_fields_js_config() );
+				wp_enqueue_script( 'wcpay-upe-checkout' );
+			} else {
+				wp_localize_script( 'WCPAY_CHECKOUT', 'wcpayConfig', WC_Payments::get_wc_payments_checkout()->get_payment_fields_js_config() );
+				wp_enqueue_script( 'WCPAY_CHECKOUT' );
+			}
```

Comment:
May simplify the code a bit. 

```suggestion

			$handle = 'WCPAY_CHECKOUT';
			$object_name = 'wcpayConfig';

			if ( WC_Payments_Features::is_upe_legacy_enabled() ) {
				$handle = 'wcpay-upe-checkout';
			} elseif ( WC_Payments_Features::is_upe_split_enabled() ) {
				$handle = 'wcpay-upe-checkout';
				$object_name = 'wcpay_upe_config';
			}

			wp_localize_script( $handle, $object_name, WC_Payments::get_wc_payments_checkout()->get_payment_fields_js_config() );
			wp_enqueue_script( $handle );
```

Not something to change here, but IMO, these $handle and $object_name should take advantage of const.

Reply:
Done.